front/src/modules/pathfinding/components/Pathfinding/PathfindingV2.tsx

```diff
@@ -0,0 +1,492 @@
+import { useSelector } from 'react-redux';
+
+import InfraLoadingState from 'applications/operationalStudies/components/Scenario/InfraLoadingState';
+import type { ManageTrainSchedulePathProperties } from 'applications/operationalStudies/types';
+import infraLogo from 'assets/pictures/components/tracks.svg';
+import { enhancedEditoastApi } from 'common/api/enhancedEditoastApi';
+import type {
+  PostV2InfraByInfraIdPathPropertiesApiArg,
+  PostV2InfraByInfraIdPathfindingBlocksApiArg,
+  RollingStockWithLiveries,
+} from 'common/api/osrdEditoastApi';
+import { osrdEditoastApi } from 'common/api/osrdEditoastApi';
+import { Spinner } from 'common/Loaders';
+import { useOsrdConfActions, useOsrdConfSelectors } from 'common/osrdContext';
+import { initialState } from 'modules/pathfinding/consts';
+import type { PathfindingActionV2, PathfindingState } from 'modules/pathfinding/types';
+import {
+  formatSuggestedOperationalPoints,
+  getPathfindingQuery,
+  insertViasInOPs,
+} from 'modules/pathfinding/utils';
+import { useStoreDataForRollingStockSelector } from 'modules/rollingStock/components/RollingStockSelector/useStoreDataForRollingStockSelector';
+import type { SuggestedOP } from 'modules/trainschedule/components/ManageTrainSchedule/types';
+import { setFailure } from 'reducers/main';
+import type { PathStep } from 'reducers/osrdconf/types';
+import { useAppDispatch } from 'store';
+import { castErrorToFailure } from 'utils/error';
+import { conditionalStringConcat, formatKmValue } from 'utils/strings';
+
+export function reducer(state: PathfindingState, action: PathfindingActionV2): PathfindingState {
+  switch (action.type) {
+    case 'PATHFINDING_STARTED': {
+      return {
+        ...state,
+        running: true,
+        done: false,
+        error: '',
+        mustBeLaunched: false,
+        cancelled: false,
+      };
+    }
+    case 'PATHFINDING_CANCELLED': {
+      return {
+        ...state,
+        running: false,
+        done: false,
+        error: '',
+        mustBeLaunched: false,
+        cancelled: true,
+      };
+    }
+    case 'PATHFINDING_FINISHED': {
+      if (state.cancelled) {
+        return {
+          ...state,
+          running: false,
+          done: false,
+          error: '',
+          mustBeLaunched: false,
+          mustBeLaunchedManually: false,
+        };
+      }
+      return {
+        ...state,
+        running: false,
+        done: true,
+        error: '',
+        mustBeLaunched: false,
+        mustBeLaunchedManually: false,
+      };
+    }
+    case 'PATHFINDING_ERROR': {
+      return {
+        ...state,
+        running: false,
+        done: false,
+        error: action.message || '',
+        mustBeLaunched: false,
+      };
+    }
+    case 'PATHFINDING_PARAM_CHANGED':
+    case 'VIAS_CHANGED':
+    case 'INFRA_CHANGED': {
+      if (
+        !action.params ||
+        state.running ||
+        (action.type === 'INFRA_CHANGED' && compact(action.params.pathSteps).length === 0)
```

Comment:
action.params.pathstep is only used here, but I think we can handle this without it. Can you remove pastSteps from ActionV2 ?
```suggestion
        (action.type === 'INFRA_CHANGED' && !origin && !destination)
```

Reply:
Done but had to use `!action.params.origin && !action.params.destination`. Can't move l.101 before the if because we don't if `action.params` is defined yet.